### PR TITLE
Add user arg for docker driver

### PR DESCRIPTION
**Pull request by logan2211**
Allow 'user' to be specified when providing arguments to the docker
driver.

### CODE REVIEW

**Comment by lwm** [comment]
Thanks @logan2211! Can you rebase and get a CI run in for this change?

**Comment by logan2211** [comment]
Done!

File: test/unit/model/v2/test_platforms_section.py [hunk 1]
```diff
@@ -126,6 +128,7 @@ def _model_platforms_docker_errors_section_data():
                     'email': int(),
                 },
             },
+            'user', str(),
```
Comment (gundalow) [inline test/unit/model/v2/test_platforms_section.py]:
```suggestion
            'user': str(),
```

**Comment by webknjaz** [comment]
@logan2211 the intent looks great! And yet, it looks like some tests need to be adjusted. Please see errors in the CI: https://travis-ci.com/ansible/molecule/jobs/176204166#L1042.

**Comment by ssbarnea** [comment]
@logan2211 Please fix conflicts so we can merge it.

**Comment by ssbarnea** [comment]
I was unable to resolve conflicts on this due to lack of push rights, but I pushed them on https://github.com/ansible/molecule/pull/1889 -- please join the revived PR.

**Pull request closed without merging**